roles/openshift_logging_elasticsearch/templates/logging-metrics-role.j2

```diff
@@ -0,0 +1,31 @@
+---
+apiVersion: v1
+kind: List
+items:
+- apiVersion: rbac.authorization.k8s.io/v1beta1
+  kind: Role
+  metadata:
+    annotations:
+      rbac.authorization.kubernetes.io/autoupdate: "true"
+    name: prometheus-metrics-viewer
+    namespace: {{ role_namespace }}
+  rules:
+  - apiGroups:
+    - metrics.openshift.io
+    resources:
+    - prometheus
+    verbs:
+    - view
+- apiVersion: rbac.authorization.k8s.io/v1beta1
+  kind: RoleBinding
+  metadata:
+    name: prometheus-metrics-viewer
+    namespace: {{ role_namespace }}
+  roleRef:
+    apiGroup: rbac.authorization.k8s.io
+    kind: Role
+    name: prometheus-metrics-viewer
+  subjects:
+  - apiGroup: rbac.authorization.k8s.io
+    kind: User
```

Comment:
This binding is wrong because `role_user` is actually a service account.  `kind: ServiceAccount` is what you want if its always going to be a SA.  Or make that a variable in the template as well.

Reply:
You may be better off using the module since its a bit more complicated:

```yaml
subjects:
- kind: ServiceAccount
  name: {{ name }}
  namespace: {{ ns }}
```

Just make sure the module does this under the covers:

`oc adm policy add-role-to-user prometheus-metrics-viewer {{ full qualified SA name }} --role-namespace={{ ns }}`

Reply:
@enj Per your suggestion, converted these to be roles and rolebindings:

```
$ oc adm policy who-can view prometheus
Namespace: logging
Verb:      view
Resource:  prometheus

Users:  admin
        system:admin
        system:serviceaccount:default:pvinstaller

Groups: system:cluster-admins
        system:masters

Error during evaluation, results may not be complete: clusterrole.rbac.authorization.k8s.io "prometheus-metrics-viewer" not found
[jeff.cantrill@jcantril origin-aggregated-logging]$ oc get roles
NAME                        KIND
prometheus-metrics-viewer   Role.v1.authorization.openshift.io
[jeff.cantrill@jcantril origin-aggregated-logging]$ oc get rolebindings
NAME                              KIND
logging-elasticsearch-view-role   RoleBinding.v1.authorization.openshift.io
prometheus-metrics-viewer         RoleBinding.v1.authorization.openshift.io
system:deployer                   RoleBinding.v1.authorization.openshift.io
system:image-builder              RoleBinding.v1.authorization.openshift.io
system:image-puller               RoleBinding.v1.authorization.openshift.io
```
Note the error which I presume is what is keeping me from pulling metrics now when I curl the endpoint with the SA token.  Can you advise what I have missed?